chart/templates/statefulset.yaml

```
@@ -139,6 +139,41 @@ spec:
         {{- if .Values.extraContainers }}
           {{- toYaml .Values.extraContainers | nindent 8 }}
         {{- end }}
+        {{- if .Values.logger.enabled }}
+        - name: logger
+          securityContext:
+            {{- toYaml .Values.logger.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          command:
+          - varnishncsa
+          args:
+          - -a
+          {{- if .Values.logger.format }}
+          - {{- printf " -F %s" (toJson .Values.logger.format) }}
+          {{- end }}
```

I was wondering if we should at least reserve the option to offer different log formats than JSON. Maybe a `.Values.logger.json` and a `.Values.logger.format` value?

---

Also, I'm unsure if this is correctly quoted. This will result in a _single_ argument containing `"-F [json thinks...]"`, whereas we'd probably need _two_ different arguments (the first just containing `-F`, the second the JSON string:

```suggestion
          {{- if .Values.logger.format }}
          - -F
          - {{ toJson .Values.logger.format | quote }}
          {{- end }}
```

❓

---

This will break the readibility for third-party tools such as DataDog for example.
`toJson` ensures conformity to the JSON format without any alterations.

Another solution could be to quote the format in the `values.yml` file, but that seems a little painful 😅 

Wdyt?